test/unit/express-tooling/authentication-middleware.ts

```diff
@@ -0,0 +1,149 @@
+import * as chai from 'chai';
+import * as chaiAsPromised from 'chai-as-promised';
+import * as sinon from 'sinon';
+import * as sinonChai from 'sinon-chai';
+
+import { Response, Request } from 'express';
+
+import {
+  authenticationMiddleware,
+  AuthenticationMiddlewareOptions
+} from '../../../src';
+
+chai.use(chaiAsPromised);
+chai.use(sinonChai);
+const expect = chai.expect;
+
+describe('express tooling', () => {
+
+  const createRequestMock = (scopes: String[]): Request => ({
+    get: (name: string) => name,
+    $$tokeninfo: {
+      scope: scopes
+    }
+  } as any as Request);
+
+  const createResponseMock = (): Response => ({
+    sendStatus: sinon.spy((status: string) => undefined)
+  } as any as Response);
+
+  describe('authenticationMiddleware', () => {
+
+    it('should throw a TypeError, if tokenInfoEndpoint is undefined', () => {
+
+      // given
+      const config = {
+        publicEndpoints: ['/public', '/healthcheck'],
+        tokenInfoEndpoint: ''
+      };
+
+      // then
+      expect(() => { authenticationMiddleware(config); }).to.throw(TypeError);
+    });
+
+    it('should call #next on public endpoint', (done) => {
+
+      // given
+      const next = sinon.spy();
+
+      const config = {
+        publicEndpoints: ['/public', '/healthcheck'],
+        tokenInfoEndpoint: '/oauth2/tokeninfo'
+      };
+      const requestMock = {
+        originalUrl: '/healthcheck',
+        headers: {},
+        ...createRequestMock([])
+      } as any as Request;
+
+      // when
+      authenticationMiddleware(config)(requestMock, createResponseMock(), next);
+
+      // then
+      setTimeout(() => {
+        // tslint:disable-next-line
+        expect(next).to.have.been.called;
```

Comment:
`expect(next).to.have.been.calledWith()` would be an alternative where do not have to disable tslint

Reply:
sadly that does not work for `not.have.been.called`...

Reply:
@bzums 

Discussion @chai issue tracker: https://github.com/chaijs/chai/issues/41
Why chai is crap 😉 : https://github.com/moll/js-must#asserting-on-property-access
TSLINT plugin (eslint already has one): https://github.com/palantir/tslint/issues/2614

In general this happen if you have property accessors with side effects 🤯